src/Sparrow/Platform/Win32/Win32MemoryQueryMethods.cs

```diff
@@ -24,23 +30,194 @@ struct PPSAPI_WORKING_SET_EX_INFORMATION
+
+        private static readonly byte[][] RelevantFilesPostFixes =
+        {
+            Encoding.ASCII.GetBytes(".voron"),
+            Encoding.ASCII.GetBytes(".buffers")
+        };
+
+        private static readonly UnmanagedBuffersPool BuffersPool = new UnmanagedBuffersPool("AddrWillCauseHardPageFault");
+
+        private static uint? _pageSize;
+        public static uint PageSize
+        {
+            get
+            {
+                if (_pageSize != null)
+                    return _pageSize.Value;
+                GetSystemInfo(out var systemInfo);
+                _pageSize = systemInfo.pageSize;
+                return _pageSize.Value;
+            }
+        }
+
+        public static DynamicJsonArray GetMaps()
+        {
+            const uint uintMaxVal = uint.MaxValue;
+            var dja = new DynamicJsonArray();
+
+            GetSystemInfo(out var systemInfo);
+
+            var procMinAddress = systemInfo.minimumApplicationAddress;
+            var procMaxAddress = systemInfo.maximumApplicationAddress;
+            var processHandle = GetCurrentProcess();
+
+            var results = new Dictionary<string, Tuple<long, long>>();
+            var filenameString = new byte[2048];
+
+            while (procMinAddress.ToInt64() < procMaxAddress.ToInt64())
+            {
+                MEMORY_BASIC_INFORMATION memoryBasicInformation;
+                VirtualQueryEx(processHandle, (byte*)procMinAddress.ToPointer(),
+                    &memoryBasicInformation, new UIntPtr((uint)sizeof(MEMORY_BASIC_INFORMATION)));
+
+                // if this memory chunk is accessible
+                if (memoryBasicInformation.Protect == (uint)MemoryProtectionConstants.PAGE_READWRITE &&
+                    memoryBasicInformation.State == (uint)MemoryStateConstants.MEM_COMMIT &&
+                    memoryBasicInformation.Type == (uint)MemoryTypeConstants.MEM_MAPPED)
+                {
+                    var regionSize = memoryBasicInformation.RegionSize.ToInt64();
+                    for (long size = uintMaxVal; size < regionSize + uintMaxVal; size += uintMaxVal)
+                    {
+                        var partLength = size > regionSize ? regionSize % uintMaxVal : uintMaxVal;
+
+                        var totalDirty = AddrWillCauseHardPageFault((byte*)memoryBasicInformation.BaseAddress.ToPointer(), (uint)partLength,
+                            performCount: true);
+                        var totalClean = partLength - totalDirty;
+                        int stringLength;
+                        fixed (byte* pFilename = filenameString)
+                        {
+                            stringLength = GetMappedFileName(processHandle, memoryBasicInformation.BaseAddress.ToPointer(), pFilename, 2048);
+
+                            if (stringLength == 0)
+                                break;
+
+                            var foundRelevantFilename = false;
```

Comment:
Make this into a function, can check this first and only if relevant do the hard page fault check